pkg/pipeline/conntrack/conntrack.go

```diff
@@ -42,14 +45,80 @@ type ConnectionTracker interface {
 	Track(flowLogs []config.GenericMap) []config.GenericMap
 }
 
-type conntrackImpl struct {
-	config api.ConnTrack
-	hasher hash.Hash
+//////////////////////////
+
+// TODO: Does connectionStore deserve a file of its own?
+
+// connectionStore provides both retrieving a connection by its hash and iterating connections sorted by their last
+// update time.
+type connectionStore struct {
 	// TODO: should the key of the map be a custom hashStrType instead of string?
-	hash2conn                 map[string]connection
+	hash2conn map[string]*list.Element
+	connList  *list.List
+}
+
+type processConnF func(connection) (shouldDelete, shouldStop bool)
+
+func (ct *connectionStore) addConnection(hashStr string, conn connection) {
+	_, ok := ct.getConnection(hashStr)
+	if ok {
+		log.Errorf("BUG. connection with hash %v already exists in store. %v", hashStr, conn)
+	}
+	e := ct.connList.PushBack(conn)
+	ct.hash2conn[hashStr] = e
+}
```

Comment:
Will this be ported later to use the common timed-cache we recently implemented?

Reply:
I think so